src/components/api.js

```diff
@@ -0,0 +1,149 @@
+export default class Api {
+  constructor({ baseUrl, authToken }) {
+    this._baseUrl = baseUrl;
+    this._authToken = authToken;
+  }
+
+  getCardList() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: {
+        authorization: this._authToken,
+      },
+    })
+      .then((res) =>
+        res.ok ? res.json() : Promise.reject(`Error: ${res.status}`)
+      )
+      .catch((err) => {
+        console.log(err);
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in `index.js`  after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.